src/thermo/PlasmaPhase.cpp

```diff
@@ -0,0 +1,236 @@
+//! @file PlasmaPhase.cpp
+
+// This file is part of Cantera. See License.txt in the top-level directory or
+// at https://cantera.org/license.txt for license and copyright information.
+
+#include "cantera/thermo/PlasmaPhase.h"
+#include "cantera/base/global.h"
+#include "cantera/numerics/funcs.h"
+#include <boost/math/special_functions/gamma.hpp>
+
+namespace Cantera {
+
+PlasmaPhase::PlasmaPhase(const std::string& inputFile,
+                         const std::string& id_)
+    : IdealGasPhase(inputFile, id_)
+    , m_isotropicShapeFactor(2.0)
```

Comment:
This is one of the more subtle "gotchas" of C++, but you need to just call the default constructor for the base class here (by omitting the call to `IdealGasPhase(inputfile, id_)`) because calls to virtual functions aren't correctly resolved to this derived class until we actually get into this constructor. So the correct way to get specialized behavior from all base classes is to call `initThermoFile` directly from this constructor.

Reply:
I still don't quite get this. But I can change it to,
```
PlasmaPhase::PlasmaPhase(const std::string& inputFile,
                         const std::string& id_)
    :  m_isotropicShapeFactor(2.0)
{
    initThemoFile(inoutFile, id_);
}
```